### PR TITLE
Update generateseed to 0.6 spec

### PR DESCRIPTION
**Pull request by shayzluf**
[Part of] #2307 

---

# Update generate seed helpers

includes:
GenerateSeed

**Write why you are making the changes in this pull request**
to comply with 0.6 spec

**Write a summary of the changes you are making**
changes in:
- beacon-chain/core/helpers/randao.go
- beacon-chain/core/helpers/randao_test.go
- shared/bytesutil/bytes.go

**Link anything that would be helpful or relevant to the reviewers**
https://github.com/ethereum/eth2.0-specs/blob/dev/specs/core/0_beacon-chain.md#generate_seed

### CODE REVIEW

**Comment by shayzluf** [comment]
added missing argument in active index function
fixed older problems in generate seed

File: beacon-chain/core/helpers/randao.go [hunk 1]
```diff
@@ -4,35 +4,42 @@ import (
...
-//   def generate_seed(state: BeaconState,
-//                  epoch: Epoch) -> Bytes32:
-//    """
-//    Generate a seed for the given ``epoch``.
-//    """
-//    return hash(
-//        get_randao_mix(state, epoch - MIN_SEED_LOOKAHED) +
-//        get_active_index_root(state, epoch)
-//    )
+// def generate_seed(state: BeaconState,
```
Comment (terencechain) [inline beacon-chain/core/helpers/randao.go]: Please format these to look the same as other `Spec pseudocode definition`

File: beacon-chain/core/helpers/randao.go [hunk 2]
```diff
@@ -4,35 +4,42 @@ import (
...
+// """
+// return hash(
+// get_randao_mix(state, epoch - MIN_SEED_LOOKAHEAD) +
+// get_active_index_root(state, epoch) +
+// int_to_bytes32(epoch)
+// )
 func GenerateSeed(state *pb.BeaconState, wantedEpoch uint64) ([32]byte, error) {
-	if wantedEpoch > params.BeaconConfig().MinSeedLookahead {
-		wantedEpoch -= params.BeaconConfig().MinSeedLookahead
+	var wantedEpochM uint64
```
Comment (terencechain) [inline beacon-chain/core/helpers/randao.go]: What `wantedEpochM` means?

Comment (prestonvanloon) [reply beacon-chain/core/helpers/randao.go]: +1

Can this just be something like this?

```go
if wantedEpoch >= params.BeaconConfig().MinSeedLookahead {
   return [32]byte{}, fmt.Errorf(...)
}
wantedEpoch -= params.BeaconConfig().MinSeedLookahead 
```

Comment (shayzluf) [reply beacon-chain/core/helpers/randao.go]: it doesnt satisfy the paseudocode ```get_randao_mix(state, epoch - MIN_SEED_LOOKAHEAD)              + get_active_index_root(state, epoch) +...``` one time we use `epoch` one time we use 'epoch - MIN_SEED_LOOKAHEAD'

Comment (shayzluf) [reply beacon-chain/core/helpers/randao.go]: will change the naming

Comment (shayzluf) [reply beacon-chain/core/helpers/randao.go]: and use your method @prestonvanloon

File: beacon-chain/core/helpers/randao.go (hunk 1 above)
Comment (shayzluf) [reply beacon-chain/core/helpers/randao.go]: thanks used godoc to check. lgtm after fix

File: beacon-chain/core/helpers/randao.go [hunk 3]
```diff
@@ -12,27 +13,30 @@ import (
 //
 // Spec pseudocode definition:
 //   def generate_seed(state: BeaconState,
-//                  epoch: Epoch) -> Bytes32:
+// 	                   epoch: Epoch) -> Bytes32:
```
Comment (terencechain) [inline beacon-chain/core/helpers/randao.go]: revert this line please

File: beacon-chain/core/helpers/randao.go [hunk 4]
```diff
@@ -12,27 +13,30 @@ import (
...
 //    """
 //    Generate a seed for the given ``epoch``.
 //    """
 //    return hash(
-//        get_randao_mix(state, epoch - MIN_SEED_LOOKAHED) +
-//        get_active_index_root(state, epoch)
+//               get_randao_mix(state, epoch - MIN_SEED_LOOKAHEAD) +
```
Comment (terencechain) [inline beacon-chain/core/helpers/randao.go]: revert without the extra space please

File: beacon-chain/core/helpers/randao.go [hunk 5]
```diff
@@ -12,27 +13,30 @@ import (
...
+//               get_active_index_root(state, epoch) +
+//               int_to_bytes32(epoch)
 //    )
 func GenerateSeed(state *pb.BeaconState, wantedEpoch uint64) ([32]byte, error) {
-	if wantedEpoch > params.BeaconConfig().MinSeedLookahead {
-		wantedEpoch -= params.BeaconConfig().MinSeedLookahead
+	if wantedEpoch < params.BeaconConfig().MinSeedLookahead {
+		return [32]byte{}, fmt.Errorf("can't generate seed for epoch: %d that is smaller then MIN_SEED_LOOKAHEAD: %d", wantedEpoch, params.BeaconConfig().MinSeedLookahead)
```
Comment (terencechain) [inline beacon-chain/core/helpers/randao.go]: don't return error here, it's not an `assertion` from spec

File: beacon-chain/core/helpers/randao.go (hunk 3 above)
Comment (shayzluf) [reply beacon-chain/core/helpers/randao.go]: done

File: beacon-chain/core/helpers/randao.go (hunk 4 above)
Comment (shayzluf) [reply beacon-chain/core/helpers/randao.go]: done

File: beacon-chain/core/helpers/randao.go (hunk 5 above)
Comment (shayzluf) [reply beacon-chain/core/helpers/randao.go]: removed

**Pull request merged**